code/modules/shuttle/shuttle.dm

```diff
@@ -587,60 +597,67 @@
 		if(move_mode & MOVE_AREA)
 			areas_to_move[old_area] = TRUE
 
-		old_turfs[place] = move_mode
-		CHECK_TICK
+		old_turfs[oldT] = move_mode
 
 	/*******************************************All onShuttleMove procs******************************************/
 
-	index = 0
-	for(var/place in old_turfs)
-		index++
-		var/turf/oldT = place
-		var/turf/newT = new_turfs[index]
-		var/move_mode = old_turfs[place]
+	for(var/i in 1 to old_turfs.len)
+		var/turf/oldT = old_turfs[i]
+		var/turf/newT = new_turfs[i]
+		var/move_mode = old_turfs[oldT]
 		if(move_mode & MOVE_CONTENTS)
-			for(var/thing in oldT)
-				var/atom/movable/moving_atom = thing
+			var/list/old_contents = oldT.contents.Copy()
+			for(var/k in 1 to old_contents.len)
+				var/atom/movable/moving_atom = old_contents[k]
```

Comment:
for (var/k in oldT) should be faster here, I believe @RemieRichards has proven that a list isn't modified while it's iterated over

Reply:
oh yeah woops, I added the copy to test a problem I was having and forgot to switch it over.